src/structurizr/model/container.py

```diff
@@ -99,41 +98,68 @@ def __init__(
         super().__init__(**kwargs)
         self.parent = parent
         self.technology = technology
-        self.components = set(components)
+        self._components = set(components)
 
         self.tags.add(Tags.CONTAINER)
 
+    @property
+    def components(self) -> Iterable[Component]:
+        """Return read-only list of child components."""
+        return [c for c in self._components]
```

Comment:
Trying out an approach to making key properties read-only to the outside world.  See what you think.

Went for list rather than generator as it's easier to use outside and they'll never be huge.

Reply:
Right now I'm undecided what's more useful, a set or a list? So either

```py
return self._components.copy()
```

or simply

```py
return list(self._components)
```